File: AutoPowerModeSwitcher/Debugging.vb
Public Class Debugging

    Private Sub Button3_Click_1(sender As Object, e As EventArgs) Handles Button3.Click
        SettingsDatabase.Show()
    End Sub

    Private Sub Button2_Click_1(sender As Object, e As EventArgs) Handles Button2.Click

        Throw New System.Exception("You have thrown an exception")
    End Sub

 

 

    Private Sub Button5_Click(sender As Object, e As EventArgs) Handles Button5.Click
        My.Settings.Reset()
        My.Settings.Save()

    End Sub

    Private Sub CheckBox1_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox1.CheckedChanged
        If CheckBox1.Checked Then
            PreBootManager.Show()
        Else
            PreBootManager.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox2_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox2.CheckedChanged
        If CheckBox2.Checked = True Then
            BootManagerUXPort.Show()
        Else
            BootManagerUXPort.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox4_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox4.CheckedChanged
        If CheckBox4.Checked = True Then
            InstallerModule.Show()

        Else
            InstallerModule.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox8_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox8.CheckedChanged
        If CheckBox8.Checked = True Then
            UserInterface.Show()
        Else
            UserInterface.Dispose()

        End If
    End Sub

    Private Sub CheckBox7_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox7.CheckedChanged
        If CheckBox7.Checked = True Then
            CoreForm.Show()
        Else
            CoreForm.Dispose()



        End If
    End Sub

    Private Sub CheckBox6_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox6.CheckedChanged
        If CheckBox8.Checked = True Then
            UserPreferencesModule.Show()
        Else
            UserPreferencesModule.Dispose()
        End If
    End Sub

    Private Sub CheckBox5_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox5.CheckedChanged
        If CheckBox8.Checked = True Then
            AutoPowerSwitcher.Show()
        Else
            AutoPowerSwitcher.Dispose()


        End If
    End Sub

    Private Sub Button6_Click(sender As Object, e As EventArgs) Handles Button6.Click
        If CheckBox9.Checked = True Then
            Panel1.Dispose()
            DebugUI.Show()
            Close()
        Else
            My.Settings.EnableExperimentalFeatures = False
            My.Settings.Save()

            Dispose()
        End If
        
    End Sub

    Private Sub CheckBox9_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox9.CheckedChanged
        Button6.Text = "Continue"

    End Sub

    Private Sub Debugging_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Panel1.Location = New Point(6, 12)
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        DebugUI.show()

    End Sub

    Private Sub CheckBox3_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox3.CheckedChanged
           If CheckBox3.Checked = True Then
            SettingsDatabase.show


        Else
       SettingsDatabase.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub LinkLabel1_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) Handles LinkLabel1.LinkClicked
        Panel1.Visible = false
    End Sub

    Private Sub LinkLabel2_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) Handles LinkLabel2.LinkClicked
        Panel1.Visible = true
    End Sub
End Class

File: AutoPowerModeSwitcher/Debugging.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace AutoPowerModeSwitcher
{
    public partial class Debugging
    {
        public Debugging()
        {
            InitializeComponent();
        }

        private void Button3_Click_1(object sender, EventArgs e)
        {
            My.MyProject.Forms.SettingsDatabase.Show();
        }

        private void Button2_Click_1(object sender, EventArgs e)
        {

            throw new Exception("You have thrown an exception");
        }





        private void Button5_Click(object sender, EventArgs e)
        {
            My.MySettingsProperty.Settings.Reset();
            My.MySettingsProperty.Settings.Save();

        }

        private void CheckBox1_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox1.Checked)
            {
                My.MyProject.Forms.PreBootManager.Show();
            }
            else
            {
                My.MyProject.Forms.PreBootManager.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox2_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox2.Checked == true)
            {
                My.MyProject.Forms.BootManagerUXPort.Show();
            }
            else
            {
                My.MyProject.Forms.BootManagerUXPort.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox4_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox4.Checked == true)
            {
                My.MyProject.Forms.InstallerModule.Show();
            }

            else
            {
                My.MyProject.Forms.InstallerModule.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox8_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox8.Checked == true)
            {
                My.MyProject.Forms.UserInterface.Show();
            }
            else
            {
                My.MyProject.Forms.UserInterface.Dispose();

            }
        }

        private void CheckBox7_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox7.Checked == true)
            {
                My.MyProject.Forms.CoreForm.Show();
            }
            else
            {
                My.MyProject.Forms.CoreForm.Dispose();



            }
        }

        private void CheckBox6_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox8.Checked == true)
            {
                My.MyProject.Forms.UserPreferencesModule.Show();
            }
            else
            {
                My.MyProject.Forms.UserPreferencesModule.Dispose();
            }
        }

        private void CheckBox5_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox8.Checked == true)
            {
                My.MyProject.Forms.AutoPowerSwitcher.Show();
            }
            else
            {
                My.MyProject.Forms.AutoPowerSwitcher.Dispose();


            }
        }

        private void Button6_Click(object sender, EventArgs e)
        {
            if (CheckBox9.Checked == true)
            {
                Panel1.Dispose();
                My.MyProject.Forms.DebugUI.Show();
                Close();
            }
            else
            {
                My.MySettingsProperty.Settings.EnableExperimentalFeatures = false;
                My.MySettingsProperty.Settings.Save();

                Dispose();
            }

        }

        private void CheckBox9_CheckedChanged(object sender, EventArgs e)
        {
            Button6.Text = "Continue";

        }

        private void Debugging_Load(object sender, EventArgs e)
        {
            Panel1.Location = new Point(6, 12);
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.DebugUI.Show();

        }

        private void CheckBox3_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox3.Checked == true)
            {
                My.MyProject.Forms.SettingsDatabase.Show();
            }


            else
            {
                My.MyProject.Forms.SettingsDatabase.Dispose();
                GC.Collect();
            }
        }

        private void LinkLabel1_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Panel1.Visible = false;
        }

        private void LinkLabel2_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Panel1.Visible = true;
        }
    }
}